records.py

```diff
@@ -209,6 +209,18 @@ def __init__(self, db_url=None):
 
         # Enable hstore if it's available.
         self._enable_hstore()
+        self.open = True
```

Comment:
why do you have self.open? self.db.close() should be sufficient for closing the database resource